kore/src/Kore/Step/SMT/Translate.hs

```diff
@@ -219,6 +223,28 @@ translatePredicate translateUninterpreted predicate =
       where
         applicationChildrenSorts = termLikeSort <$> applicationChildren
 
+    translatePredicateExists Exists { existsVariable, existsChild } =
+        case  getHook of
+            Just builtinSort
+              | builtinSort == Builtin.Bool.sort
+              -> translateExists SMT.tBool varTerm existsChild
+              | builtinSort == Builtin.Int.sort
+              -> translateExists SMT.tInt varTerm existsChild
```

Comment:
I think the SMT should also be able to easily handle sorts that are constructor-based without recursive constructors, but it's probably not worth the trouble to add that now.